### PR TITLE
Add diagnostic cargo-bdd subcommand

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add `cargo-bdd` helper with commands to list, deduplicate, and audit step definitions
- document diagnostic tooling in the design guide, user guide, and roadmap

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie`

------
https://chatgpt.com/codex/tasks/task_e_68ae1a058ecc8322b3ccb06b792953c4

## Summary by Sourcery

Add diagnostic tooling for rstest-bdd by introducing the `cargo-bdd` subcommand, updating documentation, and registering the new crate in the workspace

New Features:
- Introduce a new `cargo-bdd` helper binary with `list-steps`, `list-duplicates`, and `list-unused` subcommands for inspecting step definitions

Build:
- Add the `cargo-bdd` crate to the workspace manifest

Documentation:
- Document the diagnostic tooling in the user guide, design guide, and roadmap

Tests:
- Add library tests for duplicate detection and unused step identification
- Add CLI integration tests for the `cargo-bdd` commands

### CODE REVIEW

File: crates/cargo-bdd/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+use std::path::PathBuf;
+
+use cargo_bdd::{duplicates, steps, unused};
+
+/// Inspect `rstest-bdd` step definitions in the current crate.
+#[derive(Parser)]
+#[command(version, about)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/main.rs]: Has this now been resolved?

File: crates/cargo-bdd/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        for entry in fs::read_dir(&dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_dir() {
+                dirs.push_back(path);
+            } else if path.extension().is_some_and(|e| e == "feature") {
+                out.push(path);
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/lib.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Consider replacing the custom BFS in gather_features with the walkdir crate to simplify recursive directory traversal and properly handle symlinks or permission errors.
- You may want to dedupe the collected feature file paths before parsing to avoid re-parsing the same file when overlapping directories or redundant paths are provided.
- For duplicate detection, consider normalizing or canonicalizing step patterns (e.g. whitespace, placeholder names) before grouping so semantically identical patterns aren’t missed or mis-reported.

File: crates/cargo-bdd/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    let mut out = Vec::new();
+    for entry in WalkDir::new(path).into_iter().filter_map(Result::ok) {
+        let p = entry.path();
+        if entry.file_type().is_file() && p.extension().is_some_and(|e| e == OsStr::new("feature"))
+        {
+            out.push(fs::canonicalize(p)?);
+        }
+    }
+    Ok(out)
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/lib.rs:116]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: crates/cargo-bdd/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                }
+            }
+        }
+    }
+
+    Ok(steps()
+        .into_iter()
+        .filter(|s| !used.contains(&(*s as *const Step)))
+        .collect())
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/lib.rs:72]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: crates/cargo-bdd/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                );
+            }
+        }
+        None => {
+            Cli::command().print_help()?;
+            println!();
+        }
+    }
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/main.rs:70]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

**Comment by leynos** [comment]
@coderabbitai pause

File: crates/cargo-bdd/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        }
+        Some(Commands::Duplicates) => {
+            for group in duplicates() {
+                let first = group[0];
+                println!("{} {}", first.keyword.as_str(), first.pattern.as_str());
+                for step in group {
+                    println!("  {}:{}", step.file, step.line);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/cargo-bdd/src/main.rs]: Has this now been resolved?

File: crates/rstest-bdd/src/types.rs [hunk 7]
```diff
@@ -95,21 +98,27 @@ impl FromStr for StepKeyword {
...
-        };
-        Ok(kw)
+        match value.trim().to_ascii_lowercase().as_str() {
+            "given" => Ok(Self::Given),
+            "when" => Ok(Self::When),
+            "then" => Ok(Self::Then),
+            "and" => Ok(Self::And),
+            "but" => Ok(Self::But),
+            _ => Err(StepKeywordParseError(value.to_string())),
+        }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd/src/types.rs:108]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.
